app/src/layouts/DashboardLayout/MenuHeader/DesktopAppPromoModal/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useMemo } from "react";
+import { RQModal } from "lib/design-system/components";
+import LINKS from "config/constants/sub/links";
+import { UAParser } from "ua-parser-js";
+import { Button } from "antd";
+import { AppleFilled, WindowsFilled } from "@ant-design/icons";
+import { capitalize } from "lodash";
+import { ReactComponent as DesktopBanner } from "./desktop-banner.svg";
+import { ReactComponent as BlueUnderline } from "./blue-underline.svg";
+import { ReactComponent as LinuxIcon } from "./linux.svg";
+import { trackDesktopAppPromoClicked } from "modules/analytics/events/common/onboarding";
+import "./desktopAppPromoModal.css";
+
+enum OS {
+  MAC = "Mac OS",
+  WINDOWS = "Windows",
+}
+
+interface DesktopAppPromoModalProps {
+  open: boolean;
+  onCancel: () => void;
+}
+
+export const DesktopAppPromoModal: React.FC<DesktopAppPromoModalProps> = ({ open, onCancel }) => {
+  const downloadLinkInfo = useMemo(() => {
+    const ua = new UAParser(window.navigator.userAgent);
+    const os = ua.getOS();
+
+    switch (os.name) {
+      case OS.MAC:
+        return {
+          os: "mac",
+          icon: <AppleFilled />,
+          link: LINKS.DOWNLOAD_DESKTOP_APP.MACOS,
+        };
+
+      case OS.WINDOWS:
+        return {
+          os: "windows",
+          icon: <WindowsFilled />,
+          link: LINKS.DOWNLOAD_DESKTOP_APP.WINDOWS,
+        };
+
+      default:
```

Comment:
Mostly there will be `mac`, `windows` and `linux` os, but `linux` was kept default because os.name can be `Ubuntu`, `Linux` or `Unix` so instead of handling all conditions, I have kept it as default case since we are checking for the rest of two.

Edge case: someone might open our app on mobile but it's very rare.

Reply:
Ok, makes sense.